src/stream/src/executor/hash_join.rs

```diff
@@ -829,6 +829,17 @@ impl<K: HashKey, S: StateStore, const T: JoinTypePrimitive> HashJoinExecutor<K,
 
             if let Some(rows) = &matched_rows {
                 join_matched_join_keys.observe(rows.len() as _);
+                if rows.len() > 10000 {
```

Comment:
suggest setting to 65536

Reply:
10K already seems quite large. It can enable us to capture a wider range of values.

Reply:
No need to follow the boundaries of the bucket IMO

Reply:
Oh you are right. I thought it's 100k.